src/wasm.h

```diff
@@ -149,7 +149,8 @@ struct Literal {
     double f64;
   };
 
-  Literal() : type(WasmType::none), f64(0) {}
+  Literal() : Literal(WasmType::none) {}
+  explicit Literal(WasmType type) : type(type) { memset(&f64, 0, sizeof(f64)); }
```

Comment:
Even with explicit, it's still a concern, isn't it? Your c++ is better than mine, but I seem to recall that the type system doesn't differentiate enums from ints. I.e. `Literal(2)` could be the enum or the int32_t, couldn't it?

As all the tests pass, this must be fine currently. I'm just worried :)